contracts/ETHRegistrarController.sol

```diff
@@ -58,8 +64,15 @@ contract ETHRegistrarController is Ownable {
     }
 
     function makeCommitment(string memory name, address owner, bytes32 secret) pure public returns(bytes32) {
+        return makeCommitmentWithConfig(name, owner, secret, address(0), address(0));
+    }
+
+    function makeCommitmentWithConfig(string memory name, address owner, bytes32 secret, address resolver, address addr) pure public returns(bytes32) {
         bytes32 label = keccak256(bytes(name));
-        return keccak256(abi.encodePacked(label, owner, secret));
+        if (resolver == address(0) && addr == address(0)) {
+            return keccak256(abi.encodePacked(label, owner, secret));
+        }
+        return keccak256(abi.encodePacked(label, owner, resolver, addr, secret));
```

Comment:
It's a bit minor but should it check the case when resolver is 0 but addr is not 0?